bot/exts/moderation/modlog.py

```diff
@@ -53,24 +55,35 @@ async def upload_log(
         if attachments is None:
             attachments = []
 
-        response = await self.bot.api_client.post(
-            'bot/deleted-messages',
-            json={
-                'actor': actor_id,
-                'creation': datetime.now(timezone.utc).isoformat(),
-                'deletedmessage_set': [
-                    {
-                        'id': message.id,
-                        'author': message.author.id,
-                        'channel_id': message.channel.id,
-                        'content': message.content.replace("\0", ""),  # Null chars cause 400.
-                        'embeds': [embed.to_dict() for embed in message.embeds],
-                        'attachments': attachment,
-                    }
-                    for message, attachment in zip_longest(messages, attachments, fillvalue=[])
-                ]
+        deletedmessage_set = [
+            {
+                "id": message.id,
+                "author": message.author.id,
+                "channel_id": message.channel.id,
+                "content": message.content.replace("\0", ""),  # Null chars cause 400.
+                "embeds": [embed.to_dict() for embed in message.embeds],
+                "attachments": attachment,
             }
-        )
+            for message, attachment in zip_longest(messages, attachments, fillvalue=[])
+        ]
+
+        try:
+            response = await self.bot.api_client.post(
+                "bot/deleted-messages",
+                json={
+                    "actor": actor_id,
+                    "creation": datetime.now(timezone.utc).isoformat(),
+                    "deletedmessage_set": deletedmessage_set,
+                }
+            )
+        except ResponseCodeError as e:
+            add_breadcrumb(
+                category="api_error",
+                message=str(e),
+                level="error",
+                data=deletedmessage_set,
+            )
```

Comment:
There's a discrepancy in the documentation of this function: https://getsentry.github.io/sentry-python/api.html?highlight=add_breadcrumb#sentry_sdk.add_breadcrumb vs https://docs.sentry.io/platforms/python/enriching-events/breadcrumbs/#manual-breadcrumbs

I don't know how to test this so let's just try and fix it if it doesn't work.

Reply:
Yea, I was following the latter. I guess we can just find out.